### PR TITLE
Quickcheck property tests

### PR DESCRIPTION
**Pull request by cptrodolfox**
Solves #47 

Introduced two property tests for private functions quoteCmd and trim. As these functions were not exposed, change a little bit of the project structure. I will continue to add one more property test for mkGenericCommand and unGenericCommand.

### CODE REVIEW

**Comment by cptrodolfox** [comment]
Introduced one more property test for mkGenericCmd and unGenericCmd and solved conflicts.

**Comment by juanpaucar** [comment]
@cptrodolfox Do not forget about fixing the issue with the haddock not being able to generate the documentation for a file and another file not being totally covered by documentation

**Comment by cptrodolfox** [comment]
@juanpaucar Fix the haddock errors.

**Comment by juanpaucar** [comment]
@cptrodolfox It looks like newer versions of GHC are still failing during CI

**Comment by cptrodolfox** [comment]
There is problem with Haddock coverage and the different version of ghc.
For versions of ghc <= 8.2.2 haddock determines that module without comments for the instances declaration are not 100%. However, for versions of ghc > 8.2.2 haddock determines that the same modules have 100% coverage. This might be related to this issue https://github.com/haskell/haddock/issues/123.

File: .travis.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ script:
  - cabal build --enable-tests
  - cabal test --enable-tests
  - cabal sdist
- - cabal haddock | grep "100%" | wc -l | grep "4"
+ - cabal haddock | grep "100%" | wc -l | grep -G "[45]" # Fixes issue with different haddock coverage with different ghc versions https://github.com/haskell/haddock/issues/123
```
[the remark this answers is not in the document]
Comment (ivanperez-keera) [reply .travis.yml:43]: Sorry, I'm facing the same issue in https://github.com/ivanperez-keera/Yampa/issues/72. What is this `[45]` thing? Is this specific to your package or something that should work for other packages too? (and, if so, why?)

[the remark this answers is not in the document]
Comment (juanpaucar) [reply .travis.yml:43]: It should be improved actually. However, this was used to ensure that the 5 exposed modules would have a 100% code coverage. But, with the changes, this PR introduced it was not the case for a function not exposed except for testing.

**Pull request merged**